### PR TITLE
Simplify instructions for using ILC

### PR DESCRIPTION
**Pull request by kant2002**
Because Microsoft.DotNet.ILCompiler on Nuget now, it's reasonable to inform user that adding custom nuget feed is for daily builds of ILC

### CODE REVIEW

File: src/coreclr/nativeaot/docs/compiling.md [hunk 1]
```diff
@@ -4,18 +4,7 @@ This document explains how to compile and publish your project using Native AOT
...
-<add key="nuget" value="https://api.nuget.org/v3/index.json" />
-```
-
-If your project has no `nuget.config` file, it may be created by running
-```bash
-> dotnet new nugetconfig
-```
-
-from the project's root directory. New package sources must be added after the `<clear />` element if you decide to keep it.
+To use Native AOT with your project, you need to add a reference to the ILCompiler NuGet package containing the Native AOT compiler and runtime.
```
Comment (am11) [inline src/coreclr/nativeaot/docs/compiling.md:7]: Starting from preview 5, you don't need to do this manually. Just add `<PublishAot>true</PublishAot>` in PropertyGroup, and SDK will pull the required packs (similar to `PublishReadyToRun` or `PublishSingleFile`, we don't add their packages manually).

File: src/coreclr/nativeaot/docs/compiling.md [hunk 2]
```diff
@@ -29,6 +18,21 @@ or by adding the following element to the project file:
   </ItemGroup>
 ```
 
+## Using daily builds
+
+For using daily builds, you need to make sure the `nuget.config` file for your project contains the following package sources under the `<packageSources>` element:
```
Comment (jkotas) [inline src/coreclr/nativeaot/docs/compiling.md:23]: Adding a nuget feed alone is not sufficient to give you daily builds with the SDK integration.

It needs to something like https://github.com/dotnet/runtime/blob/main/docs/project/dogfooding.md#install-prerequisites . Maybe just refer to that doc.

File: src/coreclr/nativeaot/docs/compiling.md [hunk 3]
```diff
@@ -2,21 +2,8 @@
 
 This document explains how to compile and publish your project using Native AOT toolchain. First, please _ensure that [pre-requisites](prerequisites.md) are installed_. If you are starting a new project, you may find the [HelloWorld sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/HelloWorld/README.md) directions useful.
 
-## Add ILCompiler package reference
+Please consult [documentation](https://github.com/dotnet/docs/blob/main/docs/core/deploying/native-aot.md) for instructions how to compile application.
```
Comment (jkotas) [inline src/coreclr/nativeaot/docs/compiling.md:5]:
```suggestion
Please consult [documentation](https://github.com/dotnet/docs/blob/main/docs/core/deploying/native-aot.md) for instructions how to compile application.

The instructions remaining in this document discuss advanced topics only.
```

Comment (LakshanF) [reply src/coreclr/nativeaot/docs/compiling.md:5]: Should we also include a link to creating the [rd.xml](https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/docs/rd-xml-format.md) file here?

@agocke was thinking ways to unblock developers when they run into issues with native aot and not wait for the SDK to pick the fix. Perhaps, this doc can serve that purpose as well?

Comment (jkotas) [reply src/coreclr/nativeaot/docs/compiling.md:5]: We have several other documents on troubleshooting, including rd.xml. https://github.com/dotnet/runtime/tree/main/src/coreclr/nativeaot/docs

File: src/coreclr/nativeaot/docs/compiling.md (hunk 2 above)
Comment (kant2002) [reply src/coreclr/nativeaot/docs/compiling.md:10]: Based on my experiments to use latest ILC with Preview 5 for example, I have to add following custom target.
```
<Target Name="ForceNightlyILC" BeforeTargets="ProcessFrameworkReferences">
  <ItemGroup>
    <KnownILCompilerPack Update="Microsoft.DotNet.ILCompiler">
      <TargetFramework>net7.0</TargetFramework>
      <ILCompilerPackNamePattern>runtime.**RID**.Microsoft.DotNet.ILCompiler</ILCompilerPackNamePattern>
      <ILCompilerPackVersion>7.0.0-preview.6.22325.1</ILCompilerPackVersion>
      <ILCompilerRuntimeIdentifiers>linux-musl-x64;linux-x64;win-x64;linux-arm;win-arm64;linux-arm64</ILCompilerRuntimeIdentifiers>
    </KnownILCompilerPack>
  </ItemGroup>
</Target>
```

Simply changing `RuntimeFrameworkVersion` was not working as planned.

Comment (jkotas) [reply src/coreclr/nativeaot/docs/compiling.md:10]: Hmm, changing `RuntimeFrameworkVersion` is not sufficient to publish regular self-contained apps either. The steps in https://github.com/dotnet/runtime/blob/main/docs/project/dogfooding.md#option-2-self-contained do not work.

The problem is that changing `RuntimeFrameworkVersion` also sets the version for Microsoft.AspNetCore.App.Runtime and Microsoft.WindowsDesktop.App.Runtime. The version numbers of nightly builds of all runtime packs rarely match.

I guess we should keep recommending to use the classic package reference (`<PackageReference Include="Microsoft.DotNet.ILCompiler" Version="7.0.0-*" />`) to get the nightly builds.

**Comment by kant2002** [comment]
@jkotas it seems to be I address all your feedback now? Or am I miss something?

File: src/coreclr/nativeaot/docs/compiling.md [hunk 4]
```diff
@@ -1,10 +1,14 @@
 # Compiling with Native AOT
 
-This document explains how to compile and publish your project using Native AOT toolchain. First, please _ensure that [pre-requisites](prerequisites.md) are installed_. If you are starting a new project, you may find the [HelloWorld sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/HelloWorld/README.md) directions useful.
+Please consult [documentation](https://github.com/dotnet/docs/blob/main/docs/core/deploying/native-aot.md) for instructions how to compile amd publish application.
 
-## Add ILCompiler package reference
+If you are starting a new project, you may find the [HelloWorld sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/HelloWorld/README.md) directions useful.
```
Comment (jkotas) [inline src/coreclr/nativeaot/docs/compiling.md:6]:
```suggestion
```

Comment (kant2002) [reply src/coreclr/nativeaot/docs/compiling.md]: I though it was valuable :) but I understand that main docs should be entry point for getting started right now.

File: src/coreclr/nativeaot/docs/compiling.md [hunk 5]
```diff
@@ -1,10 +1,12 @@
 # Compiling with Native AOT
 
-This document explains how to compile and publish your project using Native AOT toolchain. First, please _ensure that [pre-requisites](prerequisites.md) are installed_. If you are starting a new project, you may find the [HelloWorld sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/HelloWorld/README.md) directions useful.
+Please consult [documentation](https://github.com/dotnet/docs/blob/main/docs/core/deploying/native-aot.md) for instructions how to compile amd publish application.
```
Comment (jkotas) [inline src/coreclr/nativeaot/docs/compiling.md:3]:
```suggestion
Please consult [documentation](https://docs.microsoft.com/dotnet/core/deploying/native-aot) for instructions how to compile and publish application.

The rest of this document covers advanced topics only.
```

File: src/coreclr/nativeaot/docs/compiling.md [hunk 6]
```diff
@@ -1,10 +1,14 @@
 # Compiling with Native AOT
 
-This document explains how to compile and publish your project using Native AOT toolchain. First, please _ensure that [pre-requisites](prerequisites.md) are installed_. If you are starting a new project, you may find the [HelloWorld sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/HelloWorld/README.md) directions useful.
+Please consult [documentation](https://docs.microsoft.com/dotnet/core/deploying/native-aot) for instructions how to compile and publish application.
 
-## Add ILCompiler package reference
+The rest of this document covers advanced topics only.
 
-To use Native AOT with your project, you need to add a reference to the ILCompiler NuGet package containing the Native AOT compiler and runtime. Make sure the `nuget.config` file for your project contains the following package sources under the `<packageSources>` element:
+
```
Comment (jkotas) [inline src/coreclr/nativeaot/docs/compiling.md:7]:
```suggestion
```

File: src/coreclr/nativeaot/docs/compiling.md [hunk 7]
```diff
@@ -1,10 +1,13 @@
 # Compiling with Native AOT
 
-This document explains how to compile and publish your project using Native AOT toolchain. First, please _ensure that [pre-requisites](prerequisites.md) are installed_. If you are starting a new project, you may find the [HelloWorld sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/HelloWorld/README.md) directions useful.
+Please consult [documentation](https://docs.microsoft.com/dotnet/core/deploying/native-aot) for instructions how to compile and publish application.
```
Comment (jkotas) [inline src/coreclr/nativeaot/docs/compiling.md:3]: Could you please also delete https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/docs/prerequisites.md ? It is no longer referenced now that this is deleted.

**Review by jkotas — APPROVED** [review]
LGTM (modulo comment). Thank you!

File: src/coreclr/nativeaot/docs/compiling.md [hunk 8]
```diff
@@ -1,10 +1,13 @@
...
 
-## Add ILCompiler package reference
+The rest of this document covers advanced topics only.
 
-To use Native AOT with your project, you need to add a reference to the ILCompiler NuGet package containing the Native AOT compiler and runtime. Make sure the `nuget.config` file for your project contains the following package sources under the `<packageSources>` element:
+
+## Using daily builds
```
Comment (am11) [inline src/coreclr/nativeaot/docs/compiling.md:8]: This section is essentially duplicating https://github.com/dotnet/installer#installers-and-binaries, which covers all kinds of `PublishXxx` scenarios (including `PublishAot`).

Comment (jkotas) [reply src/coreclr/nativeaot/docs/compiling.md:8]: I think this small duplication is ok to make the steps easier to follow. 

It is duplicated in https://github.com/dotnet/runtime/blob/main/docs/project/dogfooding.md#obtaining-nightly-builds-of-nuget-packages as well.

**Pull request merged**